test/button/button-subtle-perceptual.js

```diff
@@ -0,0 +1,67 @@
+const puppeteer = require('puppeteer');
+const visualDiff = require('visual-diff');
+
+visualDiff.run((ctx) => {
+
+	describe('d2l-button-subtle', function() {
+
+		this.timeout(10000);
+		let browser, page;
+
+		before(async() => {
+			browser = await puppeteer.launch();
+			page = await browser.newPage();
+		});
+
+		after(() => browser.close());
+
+		const runTests = () => {
+
+			it('normal', async function() {
+				const rect = await ctx.puppeteer.getRect(page, '#normal');
+				await ctx.puppeteer.screenshotAndCompare(page, this.test.fullTitle(), { clip: rect });
+			});
+
+			it('disabled', async function() {
+				const rect = await ctx.puppeteer.getRect(page, '#disabled');
+				await ctx.puppeteer.screenshotAndCompare(page, this.test.fullTitle(), { clip: rect });
+			});
+
+			it('focus', async function() {
+				await page.click('#normal');
+				const rect = await ctx.puppeteer.getRect(page, '#normal');
+				await ctx.puppeteer.screenshotAndCompare(page, this.test.fullTitle(), { clip: rect });
+			});
+
+			it('with-icon', async function() {
+				const rect = await ctx.puppeteer.getRect(page, '#with-icon');
+				await ctx.puppeteer.screenshotAndCompare(page, this.test.fullTitle(), { clip: rect });
+			});
+
+		};
+
+		describe('wide', function() {
+
+			beforeEach(async function() {
+				await page.setViewport({width: 800, height: 800, deviceScaleFactor: 2});
```

Comment:
The `deviceScaleFactor` is interesting - it is needed on Retina displays so that screenshots are not small.  Gonna need to figure out how to normalize this I think for Travis runs.